hubblestack/files/hubblestack_nova/misc.py

```diff
@@ -313,26 +313,29 @@ def root_is_only_uid_0_account(reason=''):
     return True if result.strip() == 'root' else result
 
 
-def test_mount_attrs(mount_name,attribute,check_type='hard'):
+def test_mount_attrs(mount_name, attribute, check_type='hard'):
     '''
     Ensure that a given directory is mounted with appropriate attributes
     If check_type is soft, then in absence of volume, True will be returned
     If check_type is hard, then in absence of volume, False will be returned
     '''
     # check that the path exists on system
-    command = 'test -e ' + mount_name + ' ; echo $?'
-    output = _execute_shell_command( command)
-    if output.strip() == '1':
+    command = 'test -e ' + mount_name
+    results = __salt__['cmd.run_all'](command)
```

Comment:
It looks that much of the safety of cmd.run* functions depend on __python_shell_default_ function (https://github.com/saltstack/salt/blob/develop/salt/modules/cmdmod.py). We need to make sure that python_shell is indeed False, otherwise it's not much different what we had before.

Reply:
It is. If it weren’t, Salt would be full of vulnerabilities. The reason that function exists is so we can set it to True if someone is calling `cmd.run` as the top level job, from the command line. In that case it means they already have direct access to `cmd.run` and so we don’t need to worry about injection. (If they are malicious they have all the access they need anyway). 

`__pub_jid` is a special argument that gets passed to the top level call in salt, so the cmd module uses it to detect that top level call. We never pass that argument in hubble so we’re safe.